shared-thread-utils/src/utils/promptUtils.ts

```diff
@@ -52,38 +52,74 @@ export const limitMessages = (
 	userMessage: MessageType,
 	messagesAfterQuery: MessageType[],
 	maxCharCount: number = 40000,
-) => {
-	// Construct initial messages array with the required format
+) => {	
+	const combineAdjacentMessages = (messages: MessageType[]): MessageType[] => {
+		return messages.reduce((acc: MessageType[], curr: MessageType) => {
+			if (acc.length === 0 || acc[acc.length - 1].role !== curr.role) {
+				acc.push(curr);
+			} else {
+				acc[acc.length - 1].content += '\n\n' + curr.content;
+			}
+			return acc;
+		}, []);
+	};
```

Comment:
The only part that concerns me a bit is this, mainly because I am concerned if it breaks the flows for the other models though I am testing this now. Maybe we should create this as a separate function. Will report back in a moment though